Makefile

```diff
@@ -1,13 +1,23 @@
-
 # Image URL to use all building/pushing image targets
 IMG ?= quay.io/${USER}/odh-model-controller:latest
 # ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
-ENVTEST_K8S_VERSION = 1.29
+ENVTEST_K8S_VERSION = 1.31.0
+
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
 
-ENGINE ?= docker
+# CONTAINER_TOOL defines the container tool to be used for building images.
+# Be aware that the target commands are only tested with Docker which is
+# scaffolded by default. However, you might want to replace it to use other
+# tools. (i.e. podman)
+ENGINE ?= docker # Backwards compatibility
+CONTAINER_TOOL ?= $(ENGINE)
```

Comment:
The `CONTAINER_TOOL` came from kubebuilder scaffolding. So, I wanted to keep as it will be more familiar for users using kubebuilder defaults.

I'm also keeping `ENGINE`, to be compatible with our existing workflow.